src/core/lv_obj_pos.c

```diff
@@ -878,7 +888,12 @@ bool lv_obj_area_is_visible(const lv_obj_t * obj, lv_area_t * area)
     /*The area is not on the object*/
     if(!lv_area_intersect(area, area, &obj_coords)) return false;
 
-    lv_obj_get_transformed_area(obj, area, LV_OBJ_POINT_TRANSFORM_FLAG_RECURSIVE);
+    if(!is_transformed(obj)) {
+        *area = obj->coords;
+    }
+    else {
+        lv_obj_get_transformed_area(obj, area, LV_OBJ_POINT_TRANSFORM_FLAG_RECURSIVE);
+    }
```

Comment:
Is the condition inverted by mistake? Should the `!` be removed?

Reply:
if obj and its parent has transform property, is_transformed() will be return true.
This means that the lv_obj_get_transformed_area() call is necessary.

on the other way, return false (obj and its parent doesn't have any transform property) means that we don't need to transform object's size

Reply:
Ok, thanks for clarifying